bin/common.sh

```diff
@@ -128,9 +127,70 @@ function for_each_cluster() {
   done
 }
 
-hf_templates_init() {
+function hf_templates_init() {
   local out_dir="$1"
   shift
-  [ -z "$*" ] && hf -f helmfile.tpl/helmfile-init.yaml template --skip-deps --output-dir="$out_dir" >/dev/null 2>&1
-  hf "$@" template --skip-deps --output-dir="$out_dir" >/dev/null 2>&1
+  [[ $all ]] && hf -f helmfile.tpl/helmfile-init.yaml template --skip-deps --output-dir="$out_dir" >/dev/null 2>&1
+  hf $(echo ${label:+"-l $label"} | xargs) template --skip-deps --output-dir="$out_dir" >/dev/null 2>&1
+}
+
+##############################################################################################
+# Use OPTIONS/LONGOPTS(LONGOPTIONS) to set additional parameters.
+# Please update this function comment if you add additional parameters.
+# Outputs:
+#     STDERR 2: parse_args without options
+#     STDERR 3: no '--' passed with options
+#     STDERR 4: special ERR reserved for checking if enhanced getopt is present on the host
+#     STDERR 5: no options specified
+# Returns:
+#     all -> if passed, sets to 'y' and can be used globally in conditional statements
+#     label -> if passed (e.g. label init=true), sets to label (e.g. 'init=true') and
+#              can be used globally in conditional statements
+# Resources:
+# - https://stackoverflow.com/a/29754866
+##############################################################################################
+function parse_args() {
+  if [[ "$*" != "" ]]; then
+    ! getopt --test >/dev/null
+    if [[ ${PIPESTATUS[0]} -ne 4 ]]; then
+      echo '`getopt --test` failed in this environment.'
+      exit 1
+    fi
+
+    OPTIONS=Al:
+    LONGOPTS=all,label:
+
+    # - regarding ! and PIPESTATUS see above
+    # - temporarily store output to be able to check for errors
+    # - activate quoting/enhanced mode (e.g. by writing out “--options”)
+    # - pass arguments only via   -- "$@"   to separate them correctly
+    ! PARSED=$(getopt --options=$OPTIONS --longoptions=$LONGOPTS --name "$0" -- "$@")
+    if [[ ${PIPESTATUS[0]} -ne 0 ]]; then
+      exit 2
+    fi
+    eval set -- "$PARSED"
+    while true; do
+      case "$1" in
+        -l | --label)
+          label=$2
+          shift 2
+          ;;
+        -A | --all)
+          all=y
+          shift
+          ;;
+        --)
+          shift
+          break
+          ;;
+        *)
+          echo "Programming error: expected '--' but got $1"
+          exit 3
+          ;;
+      esac
+    done
+  else
+    echo "Error: --all or --label not specified"
+    exit 5
+  fi
```

Comment:
Implementation of https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash.

My vision for this function is to extend it for all the other options we add and add them to other functions as well. This should become the source of truth for passing options. Only have to add extra options and update the function comment. 

My criticism would be that the variables are going to pollute the global namespace. We can pass them around but there isn't a function return call so I don't know the nice way of doing this.

Reply:
Good idea. One thing: can we give this function a meaningful name and NOT put it in `common.sh`? I suggest we instead start another file `bin/functions.sh` that holds all the shared functions that are NOT used by `bin/otomi`.

Reply:
`parse_args` is not meaningful? It parses args? Or parses opts (options)?